From: GA36 FW <ga36@localhost>
Subject: [PATCH] 0005 ga36 dts add DSI host D-PHY and panel

Add MIPI DSI host controller, D-PHY, and JD9366 panel nodes
to the GA36-MB V1.2 U-Boot device tree.

The DSI controller on A33 is at 0x01c24000 and uses the MIPI DSI
clock from the CCU. The panel is a 2-lane RGB888 JD9366 at 640x480.

---

 arch/arm/dts/sun8i-a33-ga36-mb-v1.2.dts | 72 ++++++++++++++++++++++++++++
 1 file changed, 72 insertions(+)

diff --git a/arch/arm/dts/sun8i-a33-ga36-mb-v1.2.dts b/arch/arm/dts/sun8i-a33-ga36-mb-v1.2.dts
index xxxxxxx..yyyyyyy 100644
--- a/arch/arm/dts/sun8i-a33-ga36-mb-v1.2.dts
+++ b/arch/arm/dts/sun8i-a33-ga36-mb-v1.2.dts
@@ -50,6 +50,78 @@
 
 #include "axp223.dtsi"
 
+&ccu {
+	mipi_dsi_clk: mipi-dsi-clk {
+		compatible = "allwinner,sun8i-a33-mipi-dsi-clk";
+		#clock-cells = <0>;
+		clocks = <&ccu CLK_MIPI_DSI>;
+	};
+};
+
+/ {
+	dsi_host: dsi-host@1c24000 {
+		compatible = "allwinner,sun8i-a33-mipi-dsi";
+		reg = <0x01c24000 0x1000>;
+		clocks = <&ccu CLK_BUS_MIPI_DSI>, <&mipi_dsi_clk>;
+		clock-names = "bus", "mod";
+		resets = <&ccu RST_BUS_MIPI_DSI>;
+		reset-names = "dsi";
+		status = "okay";
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			dsi_out: port@1 {
+				reg = <1>;
+				dsi_out_panel: endpoint {
+					remote-endpoint = <&panel_in_dsi>;
+				};
+			};
+		};
+	};
+
+	dphy: phy@1c24400 {
+		compatible = "allwinner,sun8i-a33-mipi-dphy";
+		reg = <0x01c24400 0x400>;
+		clocks = <&ccu CLK_MIPI_DSI>;
+		clock-names = "dphy";
+		#phy-cells = <0>;
+		status = "okay";
+	};
+
+	panel: panel@0 {
+		compatible = "jd9366,8inch";
+		backlight = <&backlight>;
+		power-supply = <&reg_dcdc1>;
+		status = "okay";
+
+		port {
+			panel_in_dsi: endpoint {
+				remote-endpoint = <&dsi_out_panel>;
+			};
+		};
+
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: timing0 {
+				clock-frequency = <30000000>;
+				hactive = <640>;
+				vactive = <480>;
+				hfront-porch = <40>;
+				hback-porch = <60>;
+				hsync-len = <20>;
+				vfront-porch = <8>;
+				vback-porch = <10>;
+				vsync-len = <2>;
+				flags = "DE-on-only-mode";
+			};
+		};
+	};
+
+	backlight: backlight {
+		compatible = "pwm-backlight";
+		pwm = <&pwm 0 1000000 0>;
+		brightness-levels = <0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100>;
+		default-brightness-level = <100>;
+		status = "okay";
+	};
+};
+
 &reg_dcdc1 {
 	regulator-always-on;
 	regulator-min-microvolt = <3300000>;